service/modules/dataset-module/register_dataset.py

```diff
@@ -0,0 +1,48 @@
+import mlflow
+import json
+import sys
+import os
+
+import pandas as pd
+
+from opendp.whitenoise.client import get_dataset_client
+from opendp.whitenoise.data.adapters import load_reader, load_metadata, load_dataset
+from opendp.whitenoise.sql.private_reader import PrivateReader
+from pandasql import sqldf
+
+"""
+Sample Command line:
+
+python register_dataset.py private_csv csv_details 10.0 False local_path=serverside/path/to/example.csv
+"""
+
+if __name__ == "__main__":
+    dataset_name = sys.argv[1]
+    dataset_type = sys.argv[2]
```

Comment:
The module:

dataset = load_dataset(get_dataset(dataset_name))

do stuff

For csv details:  Write your new dataset to os.path.join(os.path.dirname(csv_details path) ,released_dataset_name.csv)
Copy the previous schema to the os.path.join(os.path.dirname(csv_details path) ,released_dataset_name.yaml)

Create the new details (no passing it in the param)
Set the budget on the new details
create a result.json("released_dataset_name":"name")
register_released(new_details)

Reply:
module has been updated according to this spec